src/olympia/amo/utils.py

```diff
@@ -1279,3 +1284,49 @@ def set_max_length_on_fields_where_necessary(self):
                         field.validators.append(
                             MaxLengthValidator(field.max_length, message=message)
                         )
+
+
+def backup_storage_enabled():
+    return bool(
+        settings.GOOGLE_APPLICATION_CREDENTIALS_STORAGE
+        and settings.GOOGLE_STORAGE_REPORTED_CONTENT_BUCKET
+    )
+
+
+def backup_storage_blob(backup_file_name_remote):
+    storage_client = google_storage.Client.from_service_account_json(
+        settings.GOOGLE_APPLICATION_CREDENTIALS_STORAGE
+    )
+    bucket = storage_client.bucket(settings.GOOGLE_STORAGE_REPORTED_CONTENT_BUCKET)
+    return bucket.blob(backup_file_name_remote)
+
+
+def copy_file_to_backup_storage(local_file_path, content_type):
+    with open(local_file_path, 'rb') as f:
+        hash_ = hashlib.sha256(f.read()).hexdigest()
+    ext = guess_extension(content_type)
+    backup_file_name_remote = f'{hash_}{ext}'
+    blob = backup_storage_blob(backup_file_name_remote)
+    # If the object already exists, generation_match_precondition set to 0 will
+    # force the upload to fail. We want that, we don't need a second copy as
+    # the name is based on the hash of the file.
+    generation_match_precondition = 0
+    try:
+        blob.upload_from_filename(
+            local_file_path, if_generation_match=generation_match_precondition
+        )
+    except PreconditionFailed:
+        pass
+    return backup_file_name_remote
+
+
+def create_signed_url_for_file_backup(backup_file_name_remote):
+    blob = backup_storage_blob(backup_file_name_remote)
+    return blob.generate_signed_url(
+        expiration=datetime.timedelta(days=REPORTED_MEDIA_BACKUP_EXPIRATION_DAYS)
+    )
+
+
+def download_file_contents_from_backup_storage(backup_file_name_remote):
+    blob = backup_storage_blob(backup_file_name_remote)
+    return blob.download_as_string()
```

Comment:
I guess this can fail if the file isn't there (expired or just not uploaded), so do we want some rudimentary exception handling in here?

Reply:
We should yeah. Not sure what should happen, probably just log ?

Reply:
I don't think we can do more than log... maybe log.exception at first?  And then reduce the severity if we're getting a lot for understandable reasons

Reply:
1f1d0deb024dd3b4fa8f21bee4d957771f1458c5 but I did this at a higher level, because that's where we'll know if it makes sense to ignore the error. This function is a bit too low level to do that.